src/PhpSpreadsheet/Worksheet/Worksheet.php

```diff
@@ -3870,4 +3870,36 @@ public function applyStylesFromArray(string $coordinate, array $styleArray): boo
 
         return true;
     }
+
+
+     /**
+      * Copy a row, including style, data and height
+      *
+      * @param int $from row that will be copied
+      * @param int $to row that will receive the copy
+      * @return void
+      */
+      public function copyRow($from, $to) {
+        $this->getRowDimension($to)->setRowHeight($this->getRowDimension($from)->getRowHeight());
+        $fromRowCellIterator = $this->getRowIterator($from,$from)->current()->getCellIterator();
+        try {
+          $fromRowCellIterator->setIterateOnlyExistingCells(true);
+          foreach ($fromRowCellIterator as $fromCellIt) {
+            $fromCoordinates = $fromCellIt->getCoordinate();
+            $toCellCoordinates = $fromCellIt->getColumn().$to;
+            // Cache data as trying to save the cells as variables doesn't work
+            $style = $this->getCell($fromCoordinates)->getStyle();
+            $value = $this->getCell($fromCoordinates)->getValue();
```

Comment:
This might not be right for formulas. Let's say cell K3 has the value 10, and A1 has the value `=1+K3`. In Excel, if I now copy row 1 to row 2, cell A2 will have the value `=1+K4`. This is probably how your method should behave. It can get tricky. Rather than trying to re-invent the wheel, your loop should probably call `copyCells` for each cell that you want to copy. That will (I hope) update the formulas and take care of the styles.